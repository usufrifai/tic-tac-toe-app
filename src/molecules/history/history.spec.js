import React from 'react';
import History from './index';
import renderer from 'react-test-renderer';

describe('History table', () => {
    it('History table should be rendered', () => {
        const history =  [
            {
                'playerOne': 'Yousuf',
                'playerTwo': 'Fahma',
                'date' : '5-19-2022',
                'winner': 'Yousuf'
            },
            {
                'playerOne': 'Rifay',
                'playerTwo': 'Ayash',
                'date' : '5-19-2022',
                'winner': 'Ayash'
            }
        ];
        const component = renderer.create(
            <History 
             history={history}
            />
          );
          let tree = component.toJSON();
          expect(tree).toMatchSnapshot();
    });
  });