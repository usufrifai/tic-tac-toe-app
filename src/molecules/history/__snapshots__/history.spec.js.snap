// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History table History table should be rendered 1`] = `
<div
  className="history-table-container"
>
  <div
    className="heading"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <h3>
      Game History
    </h3>
  </div>
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          Player One
        </th>
        <th>
          Player Two
        </th>
        <th>
          Winner
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          Yousuf
        </td>
        <td>
          Fahma
        </td>
        <td>
          Yousuf
        </td>
        <td>
          5-19-2022
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          Rifay
        </td>
        <td>
          Ayash
        </td>
        <td>
          Ayash
        </td>
        <td>
          5-19-2022
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
